src/cr-image/map.cr

```diff
@@ -299,18 +299,24 @@ module CrImage
       self.class.new(new_width, new_raw)
     end
 
-    private def initial_raw_pad(pad_type, new_width, top, bottom, left, right) : Array(T)
+    private def initial_raw_pad(pad_type, new_width, top, bottom, left, right, pad_black_value) : Array(T)
       case pad_type
-      in EdgePolicy::Black then Array(T).new((top + height + bottom) * new_width) { T.zero }
-      in EdgePolicy::Repeat then Array(T).new((top + height + bottom) * new_width) do |i|
-        current_y = i // new_width
-        next T.zero if (current_y) < top || current_y >= (top + height)
+      in EdgePolicy::Black then Array(T).new((top + height + bottom) * new_width) { pad_black_value }
+      in EdgePolicy::Repeat
+        adjusted_x = -1
+        current_y = 0
+        Array(T).new((top + height + bottom) * new_width) do |i|
+          adjusted_x += 1
+          if adjusted_x == new_width
+            adjusted_x = 0
+            current_y += 1
+          end
```

Comment:
Removing the `%` operator and replacing with this addition / if statement is a surprisingly significant optimization (~21ms to ~17ms for padding with Repeat type).